nuxeo-services/nuxeo-platform-dublincore/src/test/java/org/nuxeo/ecm/platform/dublincore/service/DublinCoreStorageServiceTest.java

```diff
@@ -0,0 +1,237 @@
+@Features(CoreFeature.class)
+@Deploy("org.nuxeo.ecm.platform.dublincore")
+public class DublinCoreStorageServiceTest {
+
+    @Inject
+    protected DublinCoreStorageService storageService;
+
+    @Inject
+    protected CoreSession session;
+
+    @Test
+    public void shouldBeUpAndRunning() {
+        assertNotNull(storageService);
+    }
+
+    @Test
+    public void shouldAddOriginatingUserWhenSystemPrincipalIsUsed() {
+        DocumentModel doc = session.createDocumentModel("/", "file", "File");
+
+        Map<String, Object> originalProperties = new HashMap<>(doc.getProperties("dublincore"));
+        Set<String> expectedChangedProperties = new HashSet<>();
+        expectedChangedProperties.add("dc:creator");
+        expectedChangedProperties.add("dc:lastContributor");
+        expectedChangedProperties.add("dc:contributors");
+
+        SystemPrincipal principal = new SystemPrincipal("Myself");
+
+        Event event = getEvent(ABOUT_TO_CREATE, session, principal);
+        storageService.addContributor(doc, event);
+
+        assertDocumentPropertiesChanges(doc, originalProperties, expectedChangedProperties);
+
+        String principalName = principal.getOriginatingUser();
+
+        assertEquals(principalName, doc.getPropertyValue("dc:creator"));
+        assertEquals(principalName, doc.getPropertyValue("dc:lastContributor"));
+
+        List<String> contributors = Arrays.asList((String[]) doc.getPropertyValue("dc:contributors"));
+        assertEquals(1, contributors.size());
+        assertTrue(contributors.contains(principalName));
+    }
+
+    @Test
+    public void shouldSetContributorsAndCreatorWhenAboutToCreate() {
+        DocumentModel doc = session.createDocumentModel("/", "file", "File");
+
+        Map<String, Object> originalProperties = new HashMap<>(doc.getProperties("dublincore"));
+        Set<String> expectedChangedProperties = new HashSet<>();
+        expectedChangedProperties.add("dc:creator");
+        expectedChangedProperties.add("dc:lastContributor");
+        expectedChangedProperties.add("dc:contributors");
+
+        Event event = getEvent(ABOUT_TO_CREATE, session, session.getPrincipal());
+        storageService.addContributor(doc, event);
+
+        assertDocumentPropertiesChanges(doc, originalProperties, expectedChangedProperties);
+
+        String principalName = session.getPrincipal().getName();
+
+        assertEquals(principalName, doc.getPropertyValue("dc:creator"));
+        assertEquals(principalName, doc.getPropertyValue("dc:lastContributor"));
+
+        List<String> contributors = Arrays.asList((String[]) doc.getPropertyValue("dc:contributors"));
+        assertEquals(1, contributors.size());
+        assertTrue(contributors.contains(principalName));
+    }
+
+    @Test
+    public void shouldAddContributorWhenAlreadyCreated() {
+        DocumentModel doc = session.createDocumentModel("/", "file", "File");
+        doc.setProperty("dublincore", "creator", "Original Creator");
```

Comment:
Whenever possible please use the more modern `doc.setPropertyValue("dc:creator", ...)`